app/controllers/spotlight/resources/upload_controller.rb

```diff
@@ -14,16 +14,16 @@ class UploadController < ApplicationController
     def create
       @resource.attributes = resource_params
 
-      if @resource.save
+      if @resource.save_and_commit
         flash[:notice] = t('spotlight.resources.upload.success')
         if params["add-and-continue"]
           redirect_to new_exhibit_resources_upload_path(@resource.exhibit)
         else
-          redirect_to admin_exhibit_catalog_index_path(@resource.exhibit)
+          redirect_to admin_exhibit_catalog_index_path(@resource.exhibit, sort: :timestamp)
```

Comment:
I'm just not sure what we want the default behavior to be, so I tried to leave the existing behavior alone.